pkg/batchperresourceattr/batchperresourceattr.go

```diff
@@ -114,7 +114,7 @@ func (bt *batchMetrics) ConsumeMetrics(ctx context.Context, td pmetric.Metrics)
 		}
 
 		// Append ResourceSpan to pmetric.Metrics for this attribute value.
-		rm.MoveTo(metricsForAttr.ResourceMetrics().AppendEmpty())
+		rm.CopyTo(metricsForAttr.ResourceMetrics().AppendEmpty())
```

Comment:
@dloucasfx Thanks for finding and fixing it!

My only concern is that it'll be a significant performance hit for all the exporters. In Splunk's use case, we usually have only one attr value per batch. So most of the time, it'll result in unnecessary copying. I think we can refactor it a bit to avoid that.

We can split the for loop into two: 
1. See how many different attribute values we have, maybe by using a map of attribute values to ResourceMetrics indices `map[string][]int`, which can be utilized next.
2. If we got a map with only one key (which should be the most common outcome), we don't do any copying. Otherwise, we do the copy.

WDYT?

Reply:
yes, we can definitely optimize the code as you've suggested.
I will work on that change

Reply:
@dmitryax let me know if this is what you had in mind. PR updated.

Reply:
Yes, this is exactly what I was thinking of. Thanks!